src/RoleBooleanGroup.php

```diff
@@ -42,13 +42,18 @@ protected function fillAttributeFromRequest(NovaRequest $request, $requestAttrib
             return;
         }
 
-        $values = collect(json_decode($request[$requestAttribute], true))
+        $model->roles()->detach();
+
+        collect(json_decode($request[$requestAttribute], true))
             ->filter(static function (bool $value) {
                 return $value;
             })
             ->keys()
-            ->toArray();
+            ->map(static function ($roleName) use ($model) {
+                $role = RoleModel::where('name', $roleName)->first();
```

Comment:
I suggest to get Role's model using `app(PermissionRegistrar::class)->getRoleClass();` instead of using RoleModel alias, because it may be changed in config.

Reply:
I agree, just made the change.